src/core/Akka/Actor/ActorPath.cs

```diff
@@ -424,11 +523,13 @@ private string Join()
             {
                 offset -= p.Name.Length + 1;
                 buffer[offset] = '/';
+
                 p.Name.CopyTo(0, buffer, offset + 1, p.Name.Length);
+
                 p = p.Parent;
             }
 
-            return new string(buffer);
+            return buffer.ToString();
```

Comment:
So this method prevents an additional copy of the string buffer from being made, which is why Gen 0 GC is lower for this iteration.